docs/aws/audit/kubernetesmonitoring/rules/ecs_cluster_container_insights_enable_remediation.mdx

```diff
@@ -1,6 +1,231 @@
+3. **Enable Container Insights for ECS Clusters**: Use the ECS client to enable Container Insights.
+4. **Verify Container Insights Status**: Check the status to ensure Container Insights is enabled.
+
+By following these steps, you can programmatically ensure that ECS Container Insights are enabled for your Kubernetes clusters using Python scripts.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Log in to the AWS Management Console and navigate to the Amazon ECS console at https://console.aws.amazon.com/ecs/.
+
+2. In the navigation pane, choose Clusters.
+
+3. On the Clusters page, select the name of the cluster you want to check.
+
+4. In the Services tab, select the name of the service you want to inspect.
+
+5. In the service detail page, scroll down to the Monitoring section. If Container Insights is enabled, you will see a section titled "CloudWatch Container Insights". If this section is not present, then Container Insights is not enabled for this service.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure the AWS CLI on your local machine. You can download it from the official AWS website and configure it using the "aws configure" command. You will be prompted to enter your AWS Access Key ID, Secret Access Key, Default region name, and Default output format.
+
+2. Once the AWS CLI is set up, you can use the "aws ecs describe-clusters" command to list all the ECS clusters in your account. The command is as follows:
+
+   ```
+   aws ecs describe-clusters --clusters <your_cluster_name>
+   ```
+
+   Replace `<your_cluster_name>` with the name of your ECS cluster.
+
+3. To check if Container Insights is enabled on your ECS cluster, you need to look for the "settings" attribute in the output of the above command. If Container Insights is enabled, you will see the following in the output:
+
+   ```
+   "settings": [
+       {
+           "name": "containerInsights",
+           "value": "enabled"
+       }
+   ]
+   ```
+
+4. If you don't see the above in the output, it means that Container Insights is not enabled on your ECS cluster. You can also use a Python script with the boto3 library to automate this process. The script would use the "describe_clusters" function in the ECS client to get the same information.
+</Accordion>
+
+<Accordion title='Using Python'>
+To check if ECS Container Insights is enabled in Kubernetes using Python scripts, you can use the AWS SDK for Python (Boto3). Here are the steps:
+
+1. **Import the necessary libraries and set up AWS credentials:**
+
+    You need to import Boto3 and set up your AWS credentials (access key, secret access key, and region). You can do this in your Python script as follows:
+
+    ```python
+    import boto3
+
+    session = boto3.Session(
+        aws_access_key_id='YOUR_ACCESS_KEY',
+        aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>